### ISSUE TITLE
`Theme.name` does not match registered name

### ISSUE THREAD

**Issue by BeksOmega** [google/blockly#6186]
**Describe the bug**

When you create a new theme **that is not entirely lowercase**, the `.name` property of the theme does not match the name in the registry. The registry normalizes all names to lowercase, but the themes are not doing this.

**To Reproduce**

1. Open the playground source.
2. Edit the `start` function to include:
```js
const theme = Blockly.Theme.defineTheme('TEST', {});
console.log(theme);
console.log(Blockly.registry.getAllItems(Blockly.registry.Type.THEME));
```
3. Observe how the theme has the name 'TEST', but the item has the key 'test'

**Expected behavior**

The theme should also have the name 'test'

**Additional context**

This was found when Greg attempted to define a non-lowercase theme in the advanced playground.

**Comment by cpcallen** [comment]
Is there some reason that the registry normalises to lowercase?  To me _that_ seems like the bug…

**Comment by BeksOmega** [comment]
Closed by #6226
